packages/amplify-category-interactions/commands/interactions/add.js

```diff
@@ -0,0 +1,39 @@
+const fs = require('fs');
+
+const subcommand = 'add';
+const category = 'interactions';
+const servicesMetadata = JSON.parse(fs.readFileSync(`${__dirname}/../../provider-utils/supported-services.json`));
+
+let options;
+
+module.exports = {
+  name: subcommand,
+  run: async (context) => {
+    const { amplify } = context;
+
+    return amplify.serviceSelectionPrompt(context, category, servicesMetadata)
+      .then((result) => {
+        options = {
+          service: result.service,
+          providerPlugin: result.providerName,
+          build: true,
+        };
+        const providerController = require(`../../provider-utils/${result.providerName}/index`);
```

Comment:
Why not use `require` all the providers at the beginning instead of doing a dynamic `require` at run time. If we are doing dynamic require and there are any syntax errors in the required file, it won't be caught until we run that specific code path

Reply:
it has to be dynamic and placed here because the providerName are only known at this step